src/System.Runtime/tests/System/DoubleTests.netcoreapp.cs

```diff
@@ -141,6 +141,406 @@ public static void Parse_Span_Invalid(string value, NumberStyles style, IFormatP
+                "262603630354748115322368533599053346255754042160606228586332807443" +
+                "018924703005556787346899784768703698535494132771566221702458461669" +
+                "916553215355296238706468887866375289955928004361779017462862722733" +
+                "744717014529914330472578638646014242520247915673681950560773208853" +
+                "293843223323915646452641434007986196650406080775491621739636492640" +
+                "497383622906068758834568265867109610417379088720358034812416003767" +
+                "05491726170293986797332763671875E-319",
+                0x0000000000010800
+                );
+
+            // http://www.exploringbinary.com/gays-strtod-returns-zero-for-inputs-just-above-2-1075/
+            // A number between 2^-2074 and 2^-1075, just slightly larger than 2^-1075.
+            // It has bit 1075 set (the denormal rounding bit), followed by 2506 zeroes,
+            // followed by one bits.  It should round up to 2^-1074.
+            CheckOneDouble(
+                "2.470328229206232720882843964341106861825299013071623822127928412503" +
+                "37753635104375932649918180817996189898282347722858865463328355177969" +
+                "89819938739800539093906315035659515570226392290858392449105184435931" +
+                "80284993653615250031937045767824921936562366986365848075700158576926" +
+                "99037063119282795585513329278343384093519780155312465972635795746227" +
+                "66465272827220056374006485499977096599470454020828166226237857393450" +
+                "73633900796776193057750674017632467360096895134053553745851666113422" +
+                "37666786041621596804619144672918403005300575308490487653917113865916" +
+                "46239524912623653881879636239373280423891018672348497668235089863388" +
+                "58792562830275599565752445550725518931369083625477918694866799496832" +
+                "40497058210285131854513962138377228261454376934125320985913276672363" +
+                "28125001e-324",
+                0x0000000000000001);
+
+            CheckOneDouble("1.0e-99999999999999999999", 0.0);
+            CheckOneDouble("0e-99999999999999999999", 0.0);
+            CheckOneDouble("0e99999999999999999999", 0.0);
+        }
+
+        private static void CheckOneDouble(string s, ulong expectedBits)
+        {
+            CheckOneDouble(s, BitConverter.Int64BitsToDouble((long)expectedBits));
+        }
+
+        private static void CheckOneDouble(string s, double expected)
+        {
+            if (!double.TryParse(s, NumberStyles.Float | NumberStyles.AllowThousands, CultureInfo.InvariantCulture, out double actual)) actual = 1.0 / 0.0;
+            if (!actual.Equals(expected))
+            {
+                throw new Exception($@"
+Error for double input ""{s}""
+   expected {InvariantToString(expected)}
+   actual {InvariantToString(actual)}");
```

Comment:
If you use a Theory, then this can just become Assert.Equal, right? Since I believe your goal here is to include not only the expected and actual values, but also the input string, which would automatically be a part of the theory failure information? At the very least, you should be able to use Assert.True(actual.Equals(expected), ...).

Reply:
I'll add a comment. This is not using the `xUnit` assert because it does not display doubles or floats to an appropriate precision to allow differentiation. (same issue as https://github.com/xunit/xunit/issues/1822)

Reply:
Even so, you could just do:
```C#
Assert.True(actual.Equals(expected), $"Expected {InvariantToString(expected)}, Actual {InvariantToString(actual)}");
```

Reply:
Fixed.